lib/std/fmt.zig

```diff
@@ -988,17 +1013,17 @@ pub fn formatBuf(
         switch (options.alignment) {
             .left => {
                 try writer.writeAll(buf);
-                try writer.writeByteNTimes(options.fill, padding);
+                try writeBytesNTimes(writer, (utf(options.fill) catch " "), padding);
             },
             .center => {
                 const left_padding = padding / 2;
                 const right_padding = (padding + 1) / 2;
-                try writer.writeByteNTimes(options.fill, left_padding);
+                try writeBytesNTimes(writer, (utf(options.fill) catch " "), left_padding);
                 try writer.writeAll(buf);
-                try writer.writeByteNTimes(options.fill, right_padding);
+                try writeBytesNTimes(writer, (utf(options.fill) catch " "), right_padding);
             },
             .right => {
-                try writer.writeByteNTimes(options.fill, padding);
+                try writeBytesNTimes(writer, (utf(options.fill) catch " "), padding);
                 try writer.writeAll(buf);
             },
         }
```

Comment:
There's no reason to create the `utf` function anyway since this logic is only and always needed along a single code path:
```diff
+        var fill_buffer: [4]u8 = undefined;
+        const fill_utf8 = if (std.unicode.utf8Encode(options.fill, &fill_buffer)) |len|
+            fill_buffer[0..len]
+        else |_|
+            " ";
         switch (options.alignment) {
             .left => {
                 try writer.writeAll(buf);
-                try writeBytesNTimes(writer, (utf(options.fill) catch " "), padding);
+                try writeBytesNTimes(writer, fill_utf8, padding);
             },
             .center => {
                 const left_padding = padding / 2;
                 const right_padding = (padding + 1) / 2;
-                try writeBytesNTimes(writer, (utf(options.fill) catch " "), left_padding);
+                try writeBytesNTimes(writer, fill_utf8, left_padding);
                 try writer.writeAll(buf);
-                try writeBytesNTimes(writer, (utf(options.fill) catch " "), right_padding);
+                try writeBytesNTimes(writer, fill_utf8, right_padding);
             },
             .right => {
-                try writeBytesNTimes(writer, (utf(options.fill) catch " "), padding);
+                try writeBytesNTimes(writer, fill_utf8, padding);
                 try writer.writeAll(buf);
             },
         }
```

Reply:
Also, please consider moving `writeBytesNTimes` to `io/writer.zig`.

Reply:
Wow thanks!